libcxx/include/__algorithm/pstl_backends/openmp/fill.h

```diff
@@ -0,0 +1,73 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _LIBCPP___ALGORITHM_PSTL_BACKENDS_OPENMP_BACKEND_FILL_H
```

Comment:
After some additional consideration, I think we should not reuse `for_each`, because we can eliminate the host to device data transfer when making a specialized implementation of `fill`.